### ISSUE TITLE
v3 support

### ISSUE THREAD

**Issue by felixfbecker** [types/npm-generic-pool#8]
https://github.com/coopernurse/node-pool/issues/150

the pool methods now return promises, and the getter methods are changed to getters. v3-alpha is available under `@next` and I would like to use it. You would either make subfolders for each version or branches. What do you prefer?

**Comment by goenning** [comment]
IMHO folders are easier to maintain and navigate through multiple versions.

I think that it's also a good time to move this repo to @types org.

**Comment by felixfbecker** [comment]
👍

**Comment by blakeembrey** [comment]
@goenning I'd love to type v3, what can I do to get this repo moved? As for folders vs branches, can we use a branch for legacy `2.x` support here instead of folders?

**Comment by goenning** [comment]
@blakeembrey I can move it to @types, just need you to update the registry.

**Comment by felixfbecker** [comment]
@goenning GitHub sets up redirects, so nothing will break 😉

**Comment by goenning** [comment]
Done! 😄

**Comment by blakeembrey** [comment]
Just updated the definition to `3.x`. I created a branch called `2.x` for legacy work (I also updated that before branching). Related to what I changed: 

* `2.x` - Updated the callback signatures to work better with `strictNullChecks` and updated all the configuration, CI, `package.json`, etc.
* `3.x` - Using promise globals and made the pool constructor private so it can't be used accidentally with `new` (need to use the `createPool` function in `3.x`)
